File: comparison/sample2.cs
public class BaseClass {

    public const int DefaultFirst = -1;
    public const double DefaultSecond = double.NaN;
    public const string DefaultThird = null;

    public int First { get; }
    public double Second { get; }
    public string Third { get; }

    public BaseClass(int first = DefaultFirst, double second = DefaultSecond, string third = DefaultThird) {
        First = first;
        Second = second;
        Third = third;
    }

}

public class ChildClass : BaseClass {

    public const object DefaultFourth = null;

    public object Fourth { get; }

    public ChildClass(int first = DefaultFirst, double second = DefaultSecond, string third = DefaultThird, object fourth = DefaultFourth)
        : base(first, second, third) {
        Fourth = fourth;
    }

}

File: comparison/sample2.java
public class BaseClass {

    protected static class BuilderImpl { public int first = -1; public double second = Double.NaN; public String third = null; }

    public static class Builder extends BuilderImpl {
        public BaseClass create() { return new BaseClass(this); }
    }

    protected int first;
    protected double second;
    protected String third;

    protected BaseClass(BuilderImpl src) {
        first = src.first;
        second = src.second;
        third = src.third;
    }

    public int first() { return first; }
    public double second() { return second; }
    public String third() { return third; }

}

public class ChildClass extends BaseClass {

    protected static class BuilderImpl extends BaseClass.BuilderImpl { public Object fourth = null; }

    public static class Builder extends BuilderImpl {
        public ChildClass create() { return new ChildClass(this); }
    }

    protected Object fourth;

    protected ChildClass(BuilderImpl src) {
        super(src);
        fourth = src.fourth;
    }

    public Object fourth() { return fourth; }

}